### ISSUE TITLE
MultiStream projection with Identities does not applies events 

### ISSUE THREAD

**Issue by Astenna** [JasperFx/marten#3424]
I'm using Marten 7.23.1.

The MultiStreamProjection is registered as inline. I want to group an event that should be applied to two projections' aggregates with:

`Identities<IEvent<CustomerAddedToService>>(x => new List<Guid>() { x.StreamId, Guid.Empty });`

The full projection looks like this:

```csharp
public class MultiStreamProjectionWithIdentities : MultiStreamProjection<ServiceCustomerList, Guid>
{
    public MultiStreamProjectionWithIdentities()
    {
        ProjectionName = nameof(MultiStreamProjectionWithIdentities);

        Identity<ServiceCreated>(x => x.Id);

        // customers without assigned service are linked to Guid.Empty service 
        Identity<CustomerCreated>(x => Guid.Empty);

        Identities<IEvent<CustomerAddedToService>>(x => new List<Guid>() { x.StreamId, Guid.Empty });
    }

    public static ServiceCustomerList Create(ServiceCreated created, ServiceCustomerList current)
    {
        return new ServiceCustomerList() { ServiceId = created.Id };
    }

    public static ServiceCustomerList Apply(CustomerCreated update, ServiceCustomerList current)
    {
        return new ServiceCustomerList() { Customers = [.. current.Customers, new Customer() { Id = update.Id, Name = update.Name }] };
    }

    public static ServiceCustomerList Apply(IQuerySession session, IEvent<CustomerAddedToService> update, ServiceCustomerList current)
    {
        if (current.ServiceId == Guid.Empty)
        {
            return new ServiceCustomerList() { Customers = current.Customers.Where(x => x.Id != update.Data.CustomerId).ToList() };
        }

        var aggregatedCustomer = session.Events.AggregateStream<CustomerStream>(update.Data.CustomerId);
        return new ServiceCustomerList() { Customers = [.. current.Customers, new Customer() { Id = update.Data.CustomerId, Name = aggregatedCustomer.Name }] };
    }
}
```

 To my suprise, the events are never applied to the `ServiceCustomerList`. I can not share the entire code, however; I tried to reproduce the bug in a minimal working example. Nevertheless, to my frustration, I can not get the minimal working example to work, as it throws on startup:

`JasperFx.CodeGeneration.UnResolvableVariableException` in `>	[Exception] JasperFx.CodeGeneration.dll!JasperFx.CodeGeneration.Model.MethodFrameArranger.FindVariable(System.Type type) Line 50	C#
`

You can find the minimal (not working) example here: 
[IdentitiesNotWorking.zip](https://github.com/user-attachments/files/17041806/IdentitiesNotWorking.zip)

It would be very valuable to me to learn why my minimal working example is not working and secondly, whether I'm doing something wrong or there is a bug in `Identities` implmentation.

**Comment by jeremydmiller** [comment]
@Astenna *That* particular method, `Identities()`, does not currently understand `IEvent<T>`. Just an oversight at this point. I know this is annoying, but you'd have to use a custom `IEventSlicer` to do exactly what you're trying to do here. 

I might take a look at the repro just to see what the codegen issue is though.

**Comment by Astenna** [comment]
Thanks for quick reply!

**Comment by jeremydmiller** [comment]
Fix for `Identities` will be in 7.29. The compilation error is from trying to push in an argument for the aggregate into the static `Create()` method. That's a no go:)
